lib/_http_common.js

```diff
@@ -290,11 +289,12 @@ exports._checkIsHttpToken = checkIsHttpToken;
  *  field-vchar    = VCHAR / obs-text
  **/
 function checkInvalidHeaderChar(val) {
-  val = '' + val;
+  if (typeof val !== 'string')
+    val = '' + val;
   for (var i = 0; i < val.length; i++) {
     const ch = val.charCodeAt(i);
-    if (ch === 9) continue;
-    if (ch <= 31 || ch > 255 || ch === 127) return true;
+    if (ch !== 9 && (ch <= 31 || ch > 255 || ch === 127))
```

Comment:
@benjamingr It's a change from
```js
for (var i = 0; i < len; i++) {
  if (a) continue;
  if (b) return true;
}
```
to
```js
for (var i = 0; i < len; i++) {
  if (!a && b) return true;
}
```

There is no difference between those forms.